gateway/src/apicast/http_proxy.lua

```diff
@@ -109,6 +125,12 @@ local function forward_https_request(proxy_uri, uri, skip_https_connect)
         ngx.log(ngx.INFO, "HTTPS Proxy: Request body is bigger than client_body_buffer_size, read the content from path='", temp_file_path, "'")
 
         if temp_file_path then
+          -- Set Content-Length only if none exist
+          if not request.headers["Content-Length"] then
+              local size, err = fsize(temp_file_path)
```

Comment:
Better to move this logc to [file_reader](https://github.com/3scale/APIcast/blob/fdf5b2ef36adaf884795e12ed91aa9ce8ffba2f3/gateway/src/resty/file.lua#L13) function where it is safely read in chunks. I think what we can do here is push each chunk into a table, iterate over each chunk in the table to get the size, sum the sizes and then set that as the `Content-Length` value. I've not thought to much about the implementation yet but given each chunk should be of a [known size](https://github.com/3scale/APIcast/blob/fdf5b2ef36adaf884795e12ed91aa9ce8ffba2f3/gateway/src/resty/file.lua#L6) we might not even need to push each chunk into a table, simply do a count and multiply the size of each chunk by the length of the table - 1 and then we only ever need to calculate the size of the last chunk. Maybe @eguzki can comment on this proposal.